### DISCUSSION TITLE
ImportError: You must install pydot (`pip install pydot`) for `plot_model` to work.

### DISCUSSION THREAD

**Discussion by wshijiang** [keras-team/keras#20303] [Q&A]
ImportError: You must install pydot (`pip install pydot`) for `plot_model` to work.  
why？

**Comment by Kxrma47** [comment]
`plot_model()` does not draw the diagram itself. Keras converts the model graph through the optional Python package `pydot`, and `pydot` calls the Graphviz `dot` executable to render the image. These are optional because they are not needed for training or inference.

You normally need both pieces:

```bash
python -m pip install pydot
```

and Graphviz from your operating system. For example, Ubuntu/Debian:

```bash
sudo apt install graphviz
```

Then verify that the same environment can see both:

```bash
python -c "import pydot; print(pydot.__version__)"
dot -V
```

If `pydot` imports but `dot -V` fails, Graphviz is missing or its executable directory is not on `PATH`—a common issue on Windows and notebooks launched before the PATH change. Restart the Python process after installing it.
